### PR TITLE
Prerender: use dispatcher instead of BroadcastChannel in utils

### PR DESCRIPTION
**Pull request by noamr**
Use polling the server using `fetch` instead of `BroadcastChannel`, to avoid relying on the notion that `BroadcastChannel` is available while prerendering.
In preperation for https://github.com/WICG/nav-speculation/pull/140

### CODE REVIEW

**Comment by nyaxt** [comment]
Should we introduce some delay either in the client or the server side? IIRC, this will issue next request immediately after receiving empty `{}` response.

**Comment by noamr** [comment]
> Should we introduce some delay either in the client or the server side? IIRC, this will issue next request immediately after receiving empty `{}` response.

yea maybe 10ms in the server

File: speculation-rules/prerender/resources/utils.js [hunk 1]
```diff
@@ -145,22 +145,44 @@ function createFrame(url) {
...
+    this.#name = name;
+    this.#baseUrl = `/speculation-rules/prerender/resources/channel.py?uid=${uid}&name=${name}`;
+  }
+
+  async #poll() {
+    if (!this.#active)
+      return;
+
+    const response = await fetch(this.#baseUrl);
+    const messages = await response.json();
```
Comment (jeremyroman) [inline speculation-rules/prerender/resources/utils.js:165]: I'm somewhat confused. It looks like this is intended to be able to both send and receive on the same channel, but I don't see any logic that prevents the channel from seeing its own outgoing messages echoed back to it, which BroadcastChannel does (and callers would reasonably expect). Am I missing it?

Comment (noamr) [reply speculation-rules/prerender/resources/utils.js:165]: It's currently actually a very crude channel where the sender and the receiver can't be the same instance, and doesn't try to be general purpose. Perhaps that makes its API surface confusing. But I can iterate on it and hopefully make it work with `dispatcher`.

Comment (noamr) [reply speculation-rules/prerender/resources/utils.js:163]: Made a new more sturdy version

Comment (noamr) [reply speculation-rules/prerender/resources/utils.js]: OK, I refactored the whole thing to use the `dispatcher` infra.

**Review by nyaxt — APPROVED** [review]

**Comment by nyaxt** [comment]
@hiroshige-g FYI

**Pull request merged**